firebase-project/src/store/index.js

```diff
@@ -0,0 +1,207 @@
+import { createStore } from "vuex";
+import { auth } from "../firebase/config";
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+import {
+  signInWithPopup,
+  GoogleAuthProvider,
+  FacebookAuthProvider,
+  TwitterAuthProvider
+} from "firebase/auth";
+import { getAuth, updateProfile } from "firebase/auth";
+import {
+  getFirestore,
+  doc,
+  setDoc,
+  collection,
+  Timestamp,
+  updateDoc,
+} from "firebase/firestore";
+import { getStorage, ref, uploadBytes } from "firebase/storage";
+
+export default createStore({
+  state: {
+    user: null,
+    post: null,
+  },
+  getters: {},
+  mutations: {
+    setUser(state, payload) {
+      state.user = payload;
+      console.log("user state changed:", state.user);
+    },
+    setPost(state, payload) {
+      state.user = payload;
+      console.log("post state changed:", state.post);
+    },
+  },
+  actions: {
+    async signup(
+      { commit },
+      {
+        email,
+        password,
+        firstName,
+        lastName,
+        mobileNumber,
+        confirmPassword,
+        profilePhoto,
+      }
+    ) {
+      try {
+        // Register the user with Firebase Authentication
+        const res = await createUserWithEmailAndPassword(
```

Comment:
@mukeshsin This method createUserWithEmailAndPassword should not be in the Store file.